src/onnx/parse_if.cpp

```diff
@@ -65,13 +67,92 @@ struct parse_if : op_parser<parse_if>
 
         auto then_out_shapes = then_mdl->get_output_shapes();
         auto else_out_shapes = else_mdl->get_output_shapes();
-        if(not std::equal(then_out_shapes.begin(),
-                          then_out_shapes.end(),
-                          else_out_shapes.begin(),
-                          else_out_shapes.end()))
+
+        assert(then_out_shapes.size() == else_out_shapes.size());
+
+        // Must have the same type for both if/else blocks by onnx spec
+        // Add exception for empty constant scalars
+        if(then_out_shapes.at(0).type() != else_out_shapes.at(0).type())
         {
             MIGRAPHX_THROW("PARSE_IF: " + info.name +
-                           " then and else sub_grahps must have same output shapes!");
+                           " then and else sub_grahps must have same output type! " +
+                           then_out_shapes.at(0).type_string() + " vs " +
+                           else_out_shapes.at(0).type_string());
+        }
+
+        if(not then_out_shapes.at(0).dynamic() && not else_out_shapes.at(0).dynamic())
+        {
+            auto then_shape   = then_out_shapes.at(0).lens();
+            auto else_shape   = else_out_shapes.at(0).lens();
```

Comment:
rename to `then_lens` ? `then_shape` sounds like it refers to `migraphx::shape`.